metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableOpWrapper.java

```diff
@@ -85,11 +84,23 @@ public IcebergTableOpWrapper(final ConnectorContext connectorContext,
             final int getIcebergPartitionsTimeout = Integer.parseInt(configuration
                 .getOrDefault(HiveConfigConstants.GET_ICEBERG_PARTITIONS_TIMEOUT, "120"));
             result = future.get(getIcebergPartitionsTimeout, TimeUnit.SECONDS);
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            try {
-                future.cancel(true);
-            } catch (Exception ignored) {
-                log.warn("Failed cancelling the task that gets the partitions for an iceberg table.");
+        } catch (Exception e) {
+            if (!future.isDone()) {
+                try {
+                    future.cancel(true);
+                } catch (Exception ignored) {
+                    log.warn("Failed cancelling the task that gets the partitions for an iceberg table.");
+                }
+            }
+            if (e instanceof ExecutionException && e.getCause() != null) {
+                //
+                // On execution exception, throw the inner exception. This is added to throw these as 4xx errors
+                // instead of 5xx.
+                //
+                if (e.getCause() instanceof IllegalArgumentException) {
```

Comment:
Per microbot team, we will return 416 as error code for indicating "the request range cannot be satisfied". Not sure if this change will be the one, would you please chat with them? Also, the change will capture the IllegalArgumentException only,   I haven't found the related iceberg code change yet. In case you have it, please point me to it..

Reply:
Sure, will let them know. This change with throw 400 error.